api/models.py

```diff
@@ -179,3 +188,19 @@ def translate_fields(cls, params: dict):
         if parent:
             translated['parent'] = ObjectId(parent)
         return translated
+
+    def set_timeout_status(self, current_time):
```

Comment:
Why do we need to pass `current_time` as a parameter? Wouldn't it be better to get time inside the method?

Reply:
Because the value of `datetime.utcnow()` will be differed when the execution will reach this function. And we have already got the current time in `main` to pass it to `wait_for_node` function. I think the current time should be the same for both the functions.

Reply:
I'm trying to understand why we'd like to have time of some moment in the past rather than current time.  Could you help me with that?

Is there any reason why we need `wait_for_node` needs time taken from `main` instead of using current time? 
Why do we need `wait_for_node` and `set_timeout_status` use the same time?

Reply:
One reason being is, that we can lower the execution time if we get the current time once, and use it in both functions. I also thought that if we get the current time in `set_timeout_status` it will have the sleep time of `wait_for_node` added to it. This will raise the chances of Node status being changed to `timeout`. But I think that is the correct thing to do logically. I will change the implementation to get the current time in both functions.